src/store/compressors.rs

```diff
@@ -45,27 +53,28 @@ impl<'de> Deserialize<'de> for Compressor {
         let buf = String::deserialize(deserializer)?;
         let compressor = match buf.as_str() {
             "none" => Compressor::None,
+            #[cfg(feature = "lz4-compression")]
             "lz4" => Compressor::Lz4,
+            #[cfg(feature = "brotli-compression")]
             "brotli" => Compressor::Brotli,
+            #[cfg(feature = "snappy-compression")]
             "snappy" => Compressor::Snappy,
+            #[cfg(feature = "zstd-compression")]
+            _ if buf.starts_with("zstd") => Compressor::Zstd(
+                ZstdCompressor::deser_from_str(&buf).map_err(serde::de::Error::custom)?,
+            ),
             _ => {
-                if buf.starts_with("zstd") {
-                    Compressor::Zstd(
-                        ZstdCompressor::deser_from_str(&buf).map_err(serde::de::Error::custom)?,
-                    )
-                } else {
-                    return Err(serde::de::Error::unknown_variant(
-                        &buf,
-                        &[
-                            "none",
-                            "lz4",
-                            "brotli",
-                            "snappy",
-                            "zstd",
-                            "zstd(compression_level=5)",
-                        ],
-                    ));
-                }
+                return Err(serde::de::Error::unknown_variant(
```

Comment:
that's misleading though.

If the store is compressed with zstd but we don't have the feature flag, we will get an error message saying something like:

It is "zstd" but we expected something in "none", "lz4", "zstd"... etc.

Reply:
I will just adjust the deserialization error message to mention only those which were built-in and also point to those which could be enabled via additional feature flags.

Reply:
> I will just adjust the deserialization error message to mention only those which were built-in and also point to those which could be enabled via additional feature flags.

Ah, the error message itself is generated by Serde. I did update the list of expected variants to match what is built-in though.

Should we keep it that way, or should I switch to custom error message to include reference to those which could be enable but are currently not?

Reply:
I suggest using a serde custom error, to tell that the compression format xxx is not available and suggestion that tantivy is maybe missing a compilation flag.

Rationale:
The unknown_variant seem appropriate here, but we really want a correct error message surfaced to the user.

Reply:
Done.